packages/next/client/future/image.tsx

```diff
@@ -529,6 +533,17 @@ export default function Image({
   let rest: Partial<ImageProps> = all
 
   let loader: ImageLoaderWithConfig = defaultLoader
+
+  if ('loader' in configContext) {
+    if (typeof configContext.loader === 'function') {
+      const contextLoader = configContext.loader
```

Comment:
Lets simplify the logic here to move it down into line 549 (we can use `res.loader || configContext.loader`)